### PR TITLE
custom prompts: add intro and new context type

### PR DESCRIPTION
**Pull request by abeatrix**
This PR adds new context types to the custom prompts feature.

Users can specify what type of context to include when building their recipe:

```
export interface CodyPromptContext {
    codebase: boolean
    openTabs?: boolean
    currentDir?: boolean
    excludeSelection?: boolean
}
```
Add intro to My Prompts and button to create cody.json file in current workspace:
image

### CODE REVIEW

**Review by kalanchan — APPROVED** [review]

File: vscode/webviews/Recipes.tsx [hunk 1]
```diff
@@ -41,27 +41,49 @@ export const Recipes: React.FunctionComponent<{
...
-                    {myPromptsEnabled &&
-                        myPrompts?.map(promptID => (
+                        <>
+                            <div>
+                                <div className={styles.recipesHeader}>
+                                    <span>My Prompts</span>
+                                    <VSCodeButton
+                                        type="button"
+                                        appearance="icon"
+                                        onClick={() => onMyPromptClick('add')}
```
Comment (st0nebraker) [inline vscode/webviews/Recipes.tsx:51]: Is this the right command? I'm now getting this error when clicking `+`
image

**Review by st0nebraker — APPROVED** [review]
LGTM! I'm seeing one potential bug below

File: vscode/webviews/Recipes.tsx (hunk 1 above)
Comment (abeatrix) [reply vscode/webviews/Recipes.tsx:51]: thank you for catching this <3

File: lib/shared/src/chat/recipes/my-prompt.ts [hunk 2]
```diff
@@ -67,39 +67,42 @@ export class MyPrompt implements Recipe {
...
-
         return contextMessages.slice(-12)
     }
 
-    // Get context from current editor open tabs
+    // Get con text from current editor open tabs
     public static getEditorOpenTabsContext(): ContextMessage[] {
         const contextMessages: ContextMessage[] = []
-        // Skip the current active tab (which is already included in selection context), files in currentDir, and non-file tabs
+        // Skip th e current active tab (which is already included in selection context), files in currentDir, and non-file tabs
```
[the remark this answers is not in the document]
Comment (abeatrix) [reply lib/shared/src/chat/recipes/my-prompt.ts]: Yea it's so weird, thanks again for spotting this!

[the remark this answers is not in the document]
Comment (abeatrix) [reply lib/shared/src/chat/recipes/my-prompt.ts]: Yea it's so weird, thanks again for spotting this!

**Pull request merged**